internal/platform/open_file_windows.go

```diff
@@ -0,0 +1,70 @@
+package platform
+
+import (
+	"io/fs"
+	"os"
+	"syscall"
+	"unsafe"
+)
+
+func OpenFile(name string, flag int, perm fs.FileMode) (*os.File, error) {
+	fd, err := windowsOpenFile(name, flag|syscall.O_CLOEXEC, uint32(perm))
```

Comment:
I ran into this recently. So, here, we are opening the file, then if not, opening it again. I want to add the comment above `windowsOpenFile` which is why `syscall.O_CLOEXEC` and what case we expect this to err differently than `os.OpenFile` regardless of what the flags are (as the only change is O_CLOEXEC).